viewing_party/party.py

```diff
@@ -0,0 +1,254 @@
+        # add the movie title to the list of unique movies
+        friends_unique_movies.append(movie_title)
+
+    # return the list of movies the user's friends have watched but the user has not
+    return friends_unique_movies
+
+def get_friends_unique_watched_with_details(user_data):
+
+    # get a list of movie only theuser's friends have watched
+    users_unwatched_movies = get_friends_unique_watched(user_data)
+    friends_list = user_data["friends"]
+
+    unwatched_movies_and_details = set()
+
+    # loop over the list of movies only the user's friends have watched
+    for unwatched_movie in users_unwatched_movies:
+        unwatched_movie_title = unwatched_movie["title"]
+        # loop over the list of the user's friends
+        for friend in friends_list:
+            friends_watched_list = friend["watched"]
+            # loop over the list of that friend's watched movies
+            for movie in friends_watched_list:
+                friends_movie_title = movie["title"]
+                # if a movie from user's unwatched movie list is in a friend's watched movies list
+                #  add the movie title and details (such as host or genre) to a list of user's unwatched movies
+                if unwatched_movie_title == friends_movie_title:
+                    movie_details = tuple(movie.values())
+                    unwatched_movies_and_details.add(movie_details)
+    
+    # return the list of movies that only the user's friends have watched including relevant details
+    return unwatched_movies_and_details
+
+def get_available_recs(user_data):
+    
+    # get a list of movies and their streaming services only the user's friends have watched
+    unwatched_movies_and_hosts = get_friends_unique_watched_with_details(user_data)
+    
+    available_recommended_movies = []
+
+    # loop over the user's list of unwatched movies and their streaming services 
+    for title, host in unwatched_movies_and_hosts:
+        # if the unwatched movie is hosted on a service that the user is subscribed to,
+        #  append the movie and service details to the list of recommended movies
+        if host in user_data["subscriptions"]:
+            available_recommended_movies.append({"title": title, "host": host})
+
+    # return the list of movies that are hosted by a service the user is subscribed to that they have not watched but their friends have
+    return available_recommended_movies
+
+def get_new_rec_by_genre(user_data):
+    
+    # get a list of movies and their genres that the user's friends have watched but the user has not
+    unwatched_movies_and_genres = get_friends_unique_watched_with_details(user_data)
+    
+    # get the genre the user watches the most
+    favorite_genre = get_most_watched_genre(user_data)
+
+    recommended_movies_in_favorite_genre = []
+
+    # loop over user's friend's list of unwatched movies and their genres 
+    for movie in unwatched_movies_and_genres:
+        genre = movie[1]
```

Comment:
It's a little dangerous to assume that the thing in position 1 will be the genre. Dictionaries _do_ have an ordering, but it's the order the keys are added to the dictionary. Imagine if the movie dictionary listed the keys so that the genre was before the title in one or more of them. The semantics of lookups in the dictionary would be unchanged, but the order of the values in the dictionaries would effectively be unreliable. So it's often best to still work with dictionaries as though they are unordered.